src/app/pages/storage/modules/disks/components/replace-disk-dialog/replace-disk-dialog.component.ts

```diff
@@ -21,15 +24,20 @@ export interface ReplaceDiskDialogData {
   styleUrls: ['./replace-disk-dialog.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class ReplaceDiskDialogComponent {
+export class ReplaceDiskDialogComponent implements OnInit {
   form = this.formBuilder.group({
     replacement: ['', Validators.required],
     force: [false],
   });
 
+  unusedDisks: UnusedDisk[] = [];
+
   unusedDisks$ = this.ws.call('disk.get_unused').pipe(
+    tap((unusedDisks) => {
+      this.unusedDisks = unusedDisks;
```

Comment:
This is an unexpected side-effect - unusedDisks$ now has two responsibilities. 
You should remove both of these and a new loadUnusedDisks method in ngOnInit.